### PR TITLE
[feature](function) Add new parameters to 'trim'.

### PR DESCRIPTION
**Pull request by Mryange**
# Proposed changes

```
mysql> SELECT trim('ababccaab','ab') str;
+------+
| str  |
+------+
| cca  |
+------+
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/util/simd/vstring_function.h [hunk 1]
```diff
@@ -125,6 +125,101 @@ class VStringFunctions {
...
+            }
+        }
+        return StringRef(str.data + begin, end - begin + 1);
+    }
+
+    static StringRef trim(const StringRef& str, const StringRef& rhs) {
+        if (str.size == 0 || rhs.size == 0) {
+            return str;
+        }
+        return rtrim(ltrim(str, rhs)), rhs;
```
Comment (HappenLee) [inline be/src/util/simd/vstring_function.h:220]: ```rtrim(ltrim(str, rhs)), rhs;``` ?

```rtrim(ltrim(str, rhs), rhs);```

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/functions/function_string.cpp [hunk 2]
```diff
@@ -338,38 +338,134 @@ struct InitcapImpl {
...
+    }
+
+    static Status execute(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                          size_t result, size_t input_rows_count) {
+        const ColumnPtr column = block.get_by_position(arguments[0]).column;
+        const auto& rcol =
+                assert_cast<const ColumnConst*>(block.get_by_position(arguments[1]).column.get())
+                        ->get_data_column_ptr();
+        if (const ColumnString* col = check_and_get_column<ColumnString>(column.get())) {
+            if (auto col_right = check_and_get_column<ColumnString>(rcol.get())) {
```
Comment (zhangstar333) [inline be/src/vec/functions/function_string.cpp:412]: In my opinion, can directly assert_ cast get those column of string,

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p1

File: be/src/util/simd/vstring_function.h [hunk 3]
```diff
@@ -122,6 +122,101 @@ class VStringFunctions {
         return rtrim(ltrim(str));
     }
 
+    static StringRef rtrim(const StringRef& str, const StringRef& rhs) {
+        if (str.size == 0 || rhs.size == 0) {
+            return str;
+        }
+        if (rhs.size == 1) {
```
Comment (zclllyybb) [inline be/src/util/simd/vstring_function.h:129]: not required for this pr but may be considerable: for most situations these function to be used, the needles are likely to be constant. So we can also use SIMD bitwise compare to optimize them. just give code below a little change will make it.

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**